scripts/upgrade.d/004.php

```diff
@@ -0,0 +1,155 @@
+<?php namespace upgrade4;
+
+/*
+ * Copyright (C) 2020 Nethesis S.r.l.
+ * http://www.nethesis.it - nethserver@nethesis.it
+ *
+ * This script is part of NethServer.
+ *
+ * NethServer is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License,
+ * or any later version.
+ *
+ * NethServer is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with NethServer.  If not, see COPYING.
+ */
+
+$models = $container['storage']->listScopes('model');
+foreach ($models as $id) {
+    if(substr($id, 0, 6) == 'fanvil') {
+        $scope = new \Tancredi\Entity\Scope($id, $container['storage'], $container['logger']);
+        if($scope->metadata['version'] >= 4) {
+            continue;
+        }
+        $model = substr($id, 9, 1);
+        $is_second_generation = $model == '5' || $model == '6' || substr($id, 10, 2) == 'SG';
+        $scope->metadata['version'] = 4;
+        $scope->setVariables([
+            'cap_background_file' => '1',
+            'cap_screensaver_file' => '',
+            'screensaver_time' => '600',
+            'backlight_time' => '30',
+            'cap_contrast' => '1',
+            'contrast' => '4',
+            'cap_brightness' => $is_second_generation ? '1' : '',
+            'brightness' => $is_second_generation ? '5' : '',
+            'cap_backlight_time_blacklist' => '',
+            'cap_screensaver_time_blacklist' => '',
+        ]);
+        $container['logger']->info("Fixed background and screensaver settings for model $id");
+
+    } elseif(substr($id, 0, 15) == 'gigaset-Maxwell') {
+        $model = substr($id, 16, 1);
+        $scope = new \Tancredi\Entity\Scope($id, $container['storage'], $container['logger']);
+        if($scope->metadata['version'] >= 4) {
+            continue;
+        }
+        $scope->metadata['version'] = 4;
+        $scope->setVariables([
+            'cap_background_file' => ($model == 'T29' || $model == 'T46' || $model == 'T48' || $model == 'T54' || $model == 'T56' || $model == 'T57' || $model == 'T57' || $model == 'VP59') ? '1' : '',
+            'cap_screensaver_file' => ($model == 'T29' || $model == 'T46' || $model == 'T48' || $model == 'T54' || $model == 'T56' || $model == 'T57' || $model == 'T57' || $model == 'VP59') ? '1' : '',
```

Comment:
What to you think to apply the in_array() "compact" notation as suggested for Snom?

Here and below...